wasmd/services/scaffolder/wasmd.go

```diff
@@ -0,0 +1,82 @@
+package scaffolder
+
+import (
+	"context"
+	"errors"
+
+	"github.com/gobuffalo/genny/v2"
+	"github.com/ignite/apps/wasmd/templates/initialize"
+	"github.com/ignite/cli/ignite/pkg/placeholder"
+	"github.com/ignite/cli/ignite/pkg/validation"
+	"github.com/ignite/cli/ignite/pkg/xgenny"
+)
+
+// wasmdOptions holds options for creating a new module.
+type wasmdOptions struct {
+	// ibc true if the module is an ibc module
+	ibc bool
+
+	// params list of parameters
+	params []string
+}
+
+// WasmdOption configures Chain.
+type WasmdOption func(*wasmdOptions)
+
+// WithIBC scaffolds a module with IBC enabled.
+func WithIBC() WasmdOption {
+	return func(m *wasmdOptions) {
+		m.ibc = true
+	}
+}
+
+// WithParams scaffolds a module with params.
+func WithParams(params []string) WasmdOption {
+	return func(m *wasmdOptions) {
+		m.params = params
+	}
+}
+
+// CreateModule creates a new empty module in the scaffolded app.
+func (s Scaffolder) InitWasmd(
+	ctx context.Context,
+	tracer *placeholder.Tracer,
+	options ...WasmdOption,
+) (sm xgenny.SourceModification, err error) {
+
+	// Apply the options
+	var wasmdOpts wasmdOptions
+	for _, apply := range options {
+		apply(&wasmdOpts)
+	}
+
+	opts := &initialize.InitOptions{
+		AppName: s.modpath.Package,
+		AppPath: s.path,
+		Version: "v0.44",
```

Comment:
Maybe a TODO comment could be added to eventually make the version dynamic?

```suggestion
		Version: "v0.44", // TODO: Allow users to specify a version
```

Reply:
Would it make sense to instead use a semver version that also includes the patch version? I saw that you are calling `s.installWasm("v0.44.0")` with the patch version.

Reply:
The idea was to have a mapping between cosmos version and all the versions of supported wasmd and maybe we could have something like a subcommand that user could choose their ideal patch version or it could be just as simple as always choosing the last updated patch. if you have any idea of how we can do it or any other idea that could solve this problem please tell us.

Reply:
Makes sense 👍 

You could check the Cosmos SDK version that the blockchain app is using first to decide on the CosmWasm version that should be scaffolded. Users could optionally select a different version using a flag, which by default uses the last updated patch, as you mentioned.

Reply:
I've just added a map for the compatibility version and please let me know what you think about it.